<?xml version="1.0"?>
<launch>

    <arg name="class_colors_file_path" default="$(find eurobin_perception)/config/class_colors_taskboard.yaml"/>
    <arg name="taskboard_frame_name" default="taskboard_frame"/>
    <arg name="num_retries" default="3"/>
    <arg name="output_dir_path" default="$(find eurobin_perception)/output_data"/>
    <arg name="udp_ip" default="localhost"/>
    <arg name="udp_output_port" default="6000"/>
    <arg name="point_cloud_topic" default="/camerka/depth/color/points"/>
    <arg name="camera_info_topic" default="/camera/color/camera_info"/>
    <arg name="detector_result_topic" default="/eurobin_perception/detection_result"/>
    <arg name="object_list_pub_topic" default="/eurobin_perception/object_positions"/>
    <arg name="object_marker_pub_topic" default="/eurobin_perception/object_markers"/>
    <arg name="cropped_pc_pub_topic" default="/eurobin_perception/cropped_pc"/>
    <arg name="save_output" default="false"/>
    <arg name="debug" default="false"/>

    <node pkg="eurobin_perception" type="pose_estimator_node" name="pose_estimator"
        output="screen" respawn="false">
        <param name="class_colors_file_path" type="str" value="$(arg class_colors_file_path)"/>
        <param name="taskboard_frame_name" type="str" value="$(arg taskboard_frame_name)"/>
        <param name="num_retries" type="int" value="$(arg num_retries)"/>
        <param name="output_dir_path" type="str" value="$(arg output_dir_path)"/>
        <param name="udp_ip" type="str" value="$(arg udp_ip)"/>
        <param name="udp_output_port" type="int" value="$(arg udp_output_port)"/>
        <param name="point_cloud_topic" type="str" value="$(arg point_cloud_topic)"/>
        <param name="camera_info_topic" type="str" value="$(arg camera_info_topic)"/>
        <param name="detector_result_topic" type="str" value="$(arg detector_result_topic)"/>
        <param name="object_list_pub_topic" type="str" value="$(arg object_list_pub_topic)"/>
        <param name="object_marker_pub_topic" type="str" value="$(arg object_marker_pub_topic)"/>
        <param name="cropped_pc_pub_topic" type="str" value="$(arg cropped_pc_pub_topic)"/>
        <param name="save_output" type="bool" value="$(arg save_output)"/>
        <param name="debug" type="bool" value="$(arg debug)"/>
    </node>

</launch>
